scheduler/app/main.py

```diff
@@ -57,14 +58,22 @@ def from_treasury_with_assets(
         }
         totalbalance = augmented_total_balance.loc[start:end]
 
-        assets = {
-            a.token_symbol: PortfolioAsset(
-                allocation=a.balance / treasury.usd_total,
-                volatility=histprices[a.token_symbol]["std_dev"].mean(),
-                risk_contribution=a.risk_contribution,
-            )
-            for a in treasury.assets
-        }
+        assets = {}
+        for a in treasury.assets:
+            try:
+                assets.update(
+                    {
+                        a.token_symbol: PortfolioAsset(
+                            allocation=a.balance / treasury.usd_total,
+                            volatility=histprices[a.token_symbol]["std_dev"].mean(),
+                            risk_contribution=a.risk_contribution,
+                        )
+                    }
+                )
+            except KeyError as e:
```

Comment:
this issue is related to the original covalent issue addressed [here](https://github.com/butterdao/whip/pull/36#discussion_r891575580), its essentially the root of most of the issues.

This issue is raised when there is an inequality in the following sets: the treasury's assets and the statistics offered for them; i.e. at least one element from the set of `treasury.assets` is not an element of the statistics available for these `treasury.assets`. Again, this is a consequence of filtering out, or blacklisting, meme/spam tokens. So when the script tries to find statistics for the asset, it will raise an error.

So again I must raise this question, how do you think we should manage these meme/spam tokens which are returning outrageous `quote`s from covalent's `portfolio_v2` endpoint?

Reply:
> So again I must raise this question, how do you think we should manage these meme/spam tokens which are returning outrageous quotes from covalent's portfolio_v2 endpoint?

The blacklisting approach at the API adapter level (in the covalent lib) is definitely a good one (even if maybe improvable, see my other comment).  
What's bothering me with this piece of code is that we are defensive against a KeyError that our own code would have introduced beforehand, with a wrongly formed `augmented_token_hist_prices`. We should strive making sure that this data structure as well as the `treasury.assets` one are always in sync.

Let's look at the call stack which is building `augmented_token_hist_prices` (last call is at bottom):

```
get_token_hist_prices
augment_token_hist_prices
build_assets
get_assets.build_treasury_with_assets
Portfolio.from_treasury_with_assets
```

In `get_token_hist_prices`, we see that the `maybe_hist_prices` dict is build with the keys from `treasury.assets` (plus `'ETH'`).

Hence, it seems to me that `treasury.assets` and `augmented_token_hist_prices` keys should be in sync (same length apart if ETH is added, and storing the same token symbols). Thus, the `try/except` we are commenting on should be superfluous. Isn't it?